File: Day 1/main.py
def read_data(filename):
    with open(filename, "r", encoding="utf-8") as f:
        input_data = [list(map(int, line.split())) for line in f]
    return zip(*input_data)


def part_one(filename):
    left, right = map(sorted, read_data(filename))
    return sum(abs(l - r) for l, r in zip(left, right))


def part_two(filename):
    left, right = map(sorted, read_data(filename))
    return sum(el * right.count(el) for el in left)


if __name__ == "__main__":
    print(f"Part One: {part_one("input.txt")}")
    print(f"Part Two: {part_two("input.txt")}")

File: Day 1/main.cpp
#include <bits/stdc++.h>
#include <iostream>
#include <cmath>

std::pair<std::vector<int>, std::vector<int>> ReadData(std::string& filename) {
    std::ifstream file(filename);
    std::string line;

    std::vector<int> arr_l, arr_r;

    while (std::getline(file, line)) {
        std::istringstream iss(line);
        int left, right;

        if (iss >> left >> right) {
            arr_l.push_back(left);
            arr_r.push_back(right);
        }
    }
    return std::make_pair(arr_l, arr_r);
}

int PartOne(std::string& filename) {
    auto [arr_l, arr_r] = ReadData(filename);
    std::sort(arr_l.begin(), arr_l.end());
    std::sort(arr_r.begin(), arr_r.end());

    int accumulator = 0;
    for (int i = 0; i < arr_l.size(); i++) {
        //std::cout << arr_l[i] << " , " << arr_r[i] << std::endl;
        accumulator += std::abs(arr_l[i] - arr_r[i]);
    }

    return accumulator;
}

int PartTwo(std::string& filename) {
    auto [arr_l, arr_r] = ReadData(filename);
    std::sort(arr_l.begin(), arr_l.end());
    std::sort(arr_r.begin(), arr_r.end());

    int accumulator = 0;
    for (int i : arr_l) {
        const int count = std::count(arr_r.begin(), arr_r.end(), i);
        accumulator += i * count;
    }

    return accumulator;
}


int main() {
    std::string filename = "input.txt";
    std::cout << "Part One: " << PartOne(filename)<< std::endl;
    std::cout << "Part Two: " << PartTwo(filename)<< std::endl;
}

